server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -202,11 +202,20 @@ public enum MetadataVersion {
     // Add new fetch request version for KIP-951
     IBP_3_7_IV4(19, "3.7", "IV4", false),
 
+    // New version for the Kafka 3.8.0 release.
+    IBP_3_8_IV0(20, "3.8", "IV0", false),
+
+    //
+    // NOTE: MetadataVersions after this point are unstable and may be changed.
+    // If users attempt to use an unstable MetadataVersion, they will get an error.
+    // Please move this comment when updating the LATEST_PRODUCTION constant.
+    //
+
     // Add ELR related supports (KIP-966).
-    IBP_3_8_IV0(20, "3.8", "IV0", true),
+    IBP_3_9_IV0(21, "3.9", "IV0", true),
 
     // Introduce version 1 of the GroupVersion feature (KIP-848).
-    IBP_4_0_IV0(21, "4.0", "IV0", false);
+    IBP_4_0_IV0(22, "4.0", "IV0", true);
```

Comment:
For my understanding, what's the reason for using `true` here?

Reply:
I don't think this needs to be `true` (at the moment, at least: someone may add metadata changes in `IBP_4_0_IV0` later). I will change it to `false`.